src/components/diagrams/singleLineDiagram/single-line-diagram.js

```diff
@@ -296,6 +296,10 @@ const SingleLineDiagram = forwardRef((props, ref) => {
         []
     );
 
+    const hasSldSizeChanged = (oldWidth, newWidth, oldHeight, newHeight) => {
```

Comment:
It is considered wise to avoid double negation.
There is a negation here and one at the unique call site.
Here you could inverse the test and name the function stillSameSizes, or even sameSizes.

Reply:
I agree, fixed